packages/agoric-cli/src/deploy.js

```diff
@@ -401,89 +404,94 @@ export { bootPlugin } from ${JSON.stringify(absPath)};
           );
 
           const modulePath = pathResolve(moduleFile);
-          const mainNS = nativeEsm
-            ? await import(modulePath)
-            : esmRequire(modulePath);
+          let mainNS = await (nativeEsm && import(modulePath));
+          if (!mainNS) {
+            mainNS = esmRequire(modulePath);
+          }
           const main = mainNS.default;
           if (typeof main !== 'function') {
             console.error(
               `${moduleFile} does not have an export default function main`,
             );
-          } else {
-            await main(bootP, {
-              bundleSource: (file, options = undefined) =>
-                bundleSource(pathResolve(file), options),
-              cache,
-              makeDefaultLeader,
-              publishBundle,
-              listConnections,
-              pathResolve,
-              installUnsafePlugin,
-              /**
-               * Recursively look up names in the context of the bootstrap
-               * promise, such as:
-               *
-               * ['agoricNames', 'oracleBrand', 'USD']
-               * ['namesByAddress']
-               * ['namesByAddress', 'agoric1...']
-               * ['namesByAddress', 'agoric1...', 'depositFacet']
-               * ['wallet', 'issuer', 'IST']
-               *
-               * @param  {...string[]} namePath
-               * @returns {Promise<any>}
-               */
-              lookup: (...namePath) => {
-                if (namePath.length === 1 && Array.isArray(namePath[0])) {
-                  // Convert single array argument to a path.
-                  namePath = namePath[0];
-                }
-                if (namePath.length === 0) {
-                  return bootP;
-                }
-                const [first, ...remaining] = namePath;
+            continue;
```

Comment:
I see why allowing `continue` was crucial to fixing `await` problems. No change suggested.